src/Core/src/ImageSources/UriImageSourceService/UriImageSourceService.iOS.cs

```diff
@@ -21,20 +23,35 @@ public partial class UriImageSourceService
 
 			try
 			{
-				// TODO: use a real caching library with the URI
-				if (imageSource is not IStreamImageSource streamImageSource)
-					return null;
+				var hash = Crc64.ComputeHashString(imageSource.Uri.OriginalString);
+				var pathToImageCache = CacheDirectory + hash + ".png";
+
+				NSData? imageData;
+
+				if (IsImageCached(pathToImageCache))
+				{
+					imageData = GetCachedImage(pathToImageCache);
+				}
+				else
+				{
+					// TODO: use a real caching library with the URI
+					if (imageSource is not IStreamImageSource streamImageSource)
+						return null;
 
-				using var stream = await streamImageSource.GetStreamAsync(cancellationToken).ConfigureAwait(false);
+					using var stream = await streamImageSource.GetStreamAsync(cancellationToken).ConfigureAwait(false);
 
-				if (stream == null)
-					throw new InvalidOperationException($"Unable to load image stream from URI '{imageSource.Uri}'.");
+					if (stream == null)
+						throw new InvalidOperationException($"Unable to load image stream from URI '{imageSource.Uri}'.");
 
-				var data = NSData.FromStream(stream);
-				if (data == null)
-					throw new InvalidOperationException("Unable to load image stream data.");
+					imageData = NSData.FromStream(stream);
+
+					if (imageData == null)
+						throw new InvalidOperationException("Unable to load image stream data.");
+
+					CacheImage(imageData, pathToImageCache);
```

Comment:
I think we need a check here to ensure caching is enabled like in Android: https://github.com/dotnet/maui/blob/12f17cfc85e7ee8e7ad3a215d4ab69befb18fa5c/src/Core/src/ImageSources/UriImageSourceService/UriImageSourceService.Android.cs#L32